eclair-core/src/main/scala/fr/acinq/eclair/transactions/CommitmentSpec.scala

```diff
@@ -50,7 +63,10 @@ object CommitmentSpec {
   }
 
   def addHtlc(spec: CommitmentSpec, direction: Direction, update: UpdateAddHtlc): CommitmentSpec = {
```

Comment:
Maybe we can remove direction from here and directly receive a `DirectedHtlc`?

Reply:
We can have a type parameter like `[T <: DirectedHtlc]` that would be used in the pattern matching, the caller can specify the direction as a type: `addHtlc[IncomingHtlc](spec, update)`

Reply:
If the caller is ready to specify `IncomingHtlc` vs `OutgoingHtlc`, I think it's simpler that he creates that directly, isn't it?
For example `addHtlc(spec, IncomingHtlc(update))`. Adding a type parameter here seems unnecessary.

Reply:
Let's do this in a follow-up PR.